README.md

```diff
@@ -1,38 +1,131 @@
+# Ansible role `haproxy`
 
-A brief description of the role goes here.
+An Ansible role for PURPOSE. Specifically, the responsibilities of this role are to:
 
-Requirements
-------------
+- Install HAPROXY
+- Setup frontend and backend for OPENIO SDS
 
-Any pre-requisites that may not be covered by Ansible itself or the role should be mentioned here. For instance, if the role uses the EC2 module, it may be a good idea to mention in this section that the boto package is required.
+## Requirements
 
-Role Variables
---------------
+- Ansible 2.4+
 
-A description of the settable variables for this role should go here, including any variables that are in defaults/main.yml, vars/main.yml, and any variables that can/should be set via parameters to the role. Any variables that are read from other roles and/or the global scope (ie. hostvars, group vars, etc.) should be mentioned here as well.
+## Role Variables
 
-Dependencies
-------------
 
-A list of other roles hosted on Galaxy should go here, plus any details in regards to parameters that may need to be set for other roles, or variables that are used from other roles.
+| Variable   | Default | Comments (type)  |
+| :---       | :---    | :---             |
+| `haproxy_certbot_frontend` | `false` | ... |
+| `haproxy_conf_path` | `/etc/haproxy/haproxy.cfg` | ... |
+| `haproxy_conf_template` | `haproxy.cfg.j2` | ... |
+| `haproxy_conscience_backend_url` | `` | ... |
+| `haproxy_conscience_frontend` | `false` | ... |
+| `haproxy_conscience_frontend_bind` | `172.17.0.2` | ... |
+| `haproxy_default_mode` | `http` | ... |
+| `haproxy_default_options` | `` | ... |
+| `haproxy_defaults_errorfiles` | `` | ... |
+| `haproxy_defaults_log` | `global` | ... |
+| `haproxy_defaults_timeouts` | `` | ... |
+| `haproxy_defaults_unique_id_format` | `'%{+X}o\ %ci` | ... |
+| `haproxy_defaults_unique_id_header` | `'X-Unique-ID'` | ... |
+| `haproxy_global_ca_base` | `/etc/ssl/certs` | ... |
+| `haproxy_global_chroot` | `/var/lib/haproxy` | ... |
+| `haproxy_global_crt_base` | `/etc/ssl/private` | ... |
+| `haproxy_global_group` | `haproxy` | ... |
+| `haproxy_global_log_facility` | `local5` | ... |
+| `haproxy_global_log_level` | `debug` | ... |
+| `haproxy_global_log_socket` | `127.0.0.1` | ... |
+| `haproxy_global_ssl_default_bind_ciphers` | `ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:ECDHE-ECDSA-DES-CBC3-SHA:ECDHE-RSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA:!DSS` | ... |
```

Comment:
I'd rather put a link to the relevant source code file#line than put the whole list here, but it's your call...